src/components/auto-complete/auto-complete.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react';
+import {SBComponent, SBStateless} from 'stylable-react-component/dist/stylable-react';
+import {root} from 'wix-react-tools';
+import {Popup} from '../../';
+import {noop} from '../../utils';
+import {CaretDown} from '../drop-down/drop-down-icons';
+import {OptionList, SelectionList} from '../selection-list/selection-list';
+import style from './auto-complete.st.css';
+
+// Selected item is a string because of selection list's constraints
+// i would love to implement it like the TreeView where the reference to an object
+// was the way to point at the selected item
+
+export type FilterPredicate = (item: string, filterString: string) => boolean;
+
+export interface AutoCompleteListProps {
+    open: boolean;
+    items?: string[];
+    onChange?: (item: string) => void;
+}
+
+export interface AutoCompleteChangeEvent extends React.ChangeEvent<HTMLInputElement> {
+    value: string;
+}
+
+export const AutoCompleteList: React.SFC<AutoCompleteListProps> = SBStateless(props => {
+    if (props.open) {
+        return (
+            <div data-automation-id="AUTO_COMPLETE_LIST">
+                <SelectionList
+                    className="auto-complete-list"
+                    dataSource={props.items}
+                    onChange={props.onChange!}
+                />
+            </div>
+        );
+    }
+    return null;
+}, style);
+
+export interface AutoCompleteProps extends React.InputHTMLAttributes<HTMLInputElement>, Partial<OptionList> {
+    open?: boolean;
+    value?: string;
+    onChange?: (event: Partial<AutoCompleteChangeEvent>) => void;
+    filter?: FilterPredicate;
+}
+
+export interface AutoCompleteState {
+    isOpen: boolean;
```

Comment:
need to consider using this totally as a controlled prop

Reply:
This comment should be properly addressed. I also believe "isOpen" should be simple controlled property.